functions/v2/ocr/app/test/index.test.js

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+require('..');
+
+const assert = require('assert');
+const sinon = require('sinon');
+const {CloudEvent} = require('cloudevent');
+
+const {Storage} = require('@google-cloud/storage');
+const storage = new Storage();
+
+const bucketName = process.env.FUNCTIONS_BUCKET;
+const filename = 'wakeupcat.jpg';
+const text = 'Wake up human!';
+const lang = 'en';
+
+const {RESULT_BUCKET} = process.env;
+
+const supertest = require('supertest');
+const {getTestServer} = require('@google-cloud/functions-framework/testing');
+
+const errorMsg = (name, propertyName) => {
+  propertyName = propertyName || name.toLowerCase();
+  return `${name} not provided. Make sure you have a "${propertyName}" property in your request`;
+};
+
+let stubCons;
+
+const stubConsole = function () {
+  stubCons = {
+    error: sinon.stub(console, 'error'),
+    log: sinon.stub(console, 'log'),
+  };
+};
+
+const restoreConsole = function () {
+  console.log.restore();
+  console.error.restore();
+};
+
+beforeEach(stubConsole);
+afterEach(restoreConsole);
+
+describe('processImage', () => {
+  describe('functions_ocr_process', () => {
+    it('processImage validates parameters', async () => {
+      const cloudEvent = CloudEvent({
+        data: {},
+      });
+      const server = getTestServer('processImage');
+      await supertest(server)
+        .post('/')
+        .send(cloudEvent)
+        .expect(500)
+        .expect(errorMsg('Bucket'));
+    });
+  });
+  describe('functions_ocr_process functions_ocr_detect', () => {
+    it('processImage detects text', async () => {
+      const cloudEvent = CloudEvent({
+        data: {
+          bucket: bucketName,
+          name: filename,
+        },
+      });
+
+      const server = getTestServer('processImage');
+      await supertest(server).post('/').send(cloudEvent);
+      assert.ok(
+        stubCons.log.calledWith(`Looking for text in image ${filename}`)
+      );
+      assert.ok(stubCons.log.calledWith('Extracted text from image:', text));
```

Comment:
Nit: instead of `stubCons`, can we use the global `console` object here?

Reply:
Oh woops yes that was me testing something out and forgot to remove it.